File: Android/app/src/main/java/catglo/com/deliveryDatabase/AddressInfo.java
package catglo.com.deliveryDatabase;

import catglo.com.deliverydroid.data.MyGeoPoint;

import java.io.Serializable;



public class AddressInfo implements Serializable{
	private static final long serialVersionUID = 1L;
	public String address;
	public MyGeoPoint location=null;
	public String phoneNumber;
	public Order associatedOrder;
	
	@Override
	public String toString(){
		if (address!=null) return address;
		return phoneNumber;
	}
}

File: Android/app/src/main/java/catglo/com/deliveryDatabase/AddressInfo.kt
package catglo.com.deliveryDatabase

import catglo.com.deliverydroid.data.MyGeoPoint

import java.io.Serializable


class AddressInfo(var address: String,
                  var location: MyGeoPoint? = null,
                  var placeId: String? = null,
                  var associatedOrder: Order? = null)
    : Serializable
{
    var phoneNumber: String = ""


    override fun toString(): String {
        return if (!address.isBlank()) address else phoneNumber
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}